### PR TITLE
StackRendererTest should remove the used window before each test

### PR DESCRIPTION
**Pull request by vogella**
During debugging of this test I realized that we never cleanup the model elements after a test, add this to the test method.

### CODE REVIEW

**Review by marcushoepfner — APPROVED** [review]

**Comment by marcushoepfner** [comment]
After having written tests in StackRendererTest in #689, I'm able to understand and review this PR.
LGTM

**Comment by vogella** [comment]
Thanks @marcushoepfner IIRC is noticed this while looking for potential places for unit tests for the onboarding stuff. Version update is done in [ 692](https://github.com/eclipse-platform/eclipse.platform.ui/pull/692)

**Comment by marcushoepfner** [comment]
Can this be merged?

**Comment by iloveeclipse** [comment]
> Can this be merged?

This PR shows 10 commits and 60 changed files. I guess something is wrong here. Can we have one commit **rebased** on top of master?

**Comment by vogella** [comment]
Yes, not sure that happened here. Will update or abondon.

**Comment by BeckerWdf** [comment]
> Yes, not sure that happened here. Will update or abondon.

@vogella: Any updates here?

File: tests/org.eclipse.e4.ui.tests/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRendererTest.java
Comment (jukzi) [inline tests/org.eclipse.e4.ui.tests/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRendererTest.java:99]: @laeubi why does github think this is JRE 11 source? (line 173)

**Comment by BeckerWdf** [comment]
@vogella: What's the state of this PR?

**Comment by BeckerWdf** [comment]
> @vogella: What's the state of this PR?

@vogella: Do you wanna finish this one?

**Comment by marcushoepfner** [comment]
I have rebased. Let's wait for the results and merge this afterwards.
Ok from your side @vogella ?

**Comment by vogella** [comment]
Thanks @marcushoepfner and @hannesN for finishing this one. The change is (of course) fine with me.

**Comment by HeikoKlare** [comment]
Was it intended that this PR is merged despite test failures on MacOS? Subsequent PRs show test failures in `StackRendererTest` and `TabStateHandlerTest` (which also uses a `StackRendererTest` internally) in the `cleanUp()`  method, which has been changed by this PR. See, e.g. https://github.com/eclipse-platform/eclipse.platform.ui/pull/2337/checks?check_run_id=30870517513

The final GitHub Actions run for MacOS of this PR before merging shows the same test failures in the logs (https://github.com/eclipse-platform/eclipse.platform.ui/actions/runs/11103802200/job/30846537413), so I guess they were introduced by this PR.
This is one of multiple according stack traces in the logs:
```
org.eclipse.swt.SWTException: Widget is disposed
	at org.eclipse.swt.SWT.error(SWT.java:4922)
	at org.eclipse.swt.SWT.error(SWT.java:4837)
	at org.eclipse.swt.SWT.error(SWT.java:4808)
	at org.eclipse.swt.widgets.Widget.error(Widget.java:853)
	at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:632)
	at org.eclipse.swt.widgets.Control.getParent(Control.java:1846)
	at org.eclipse.swt.widgets.Shell.getShells(Shell.java:1190)
	at org.eclipse.swt.widgets.Shell.updateParent(Shell.java:2269)
	at org.eclipse.swt.widgets.Shell.setWindowVisible(Shell.java:2135)
	at org.eclipse.swt.widgets.Shell.open(Shell.java:1386)
	at org.eclipse.jface.window.Window.open(Window.java:795)
	at org.eclipse.jface.dialogs.ErrorDialog.open(ErrorDialog.java:347)
	at org.eclipse.e4.ui.internal.workbench.swt.WorkbenchStatusReporter.openDialog(WorkbenchStatusReporter.java:124)
	at org.eclipse.e4.ui.internal.workbench.swt.WorkbenchStatusReporter.report(WorkbenchStatusReporter.java:62)
	at org.eclipse.e4.core.services.statusreporter.StatusReporter.show(StatusReporter.java:133)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5$1.eventLoopException(PartRenderingEngine.java:1128)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.handle(PartRenderingEngine.java:1171)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.lambda$1(PartRenderingEngine.java:1146)
	at org.eclipse.swt.internal.ExceptionStash.stash(ExceptionStash.java:63)
	at org.eclipse.swt.widgets.Widget.release(Widget.java:1432)
	at org.eclipse.swt.widgets.Control.release(Control.java:2938)
	at org.eclipse.swt.widgets.Widget.dispose(Widget.java:730)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.disposeWidget(SWTPartRenderer.java:186)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeRemoveGui(PartRenderingEngine.java:934)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:857)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.removeGui(PartRenderingEngine.java:841)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.subscribeTopicToBeRendered(PartRenderingEngine.java:184)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:56)
	at org.eclipse.e4.core.di.internal.extensions.EventObjectSupplier$DIEventHandler.handleEvent(EventObjectSupplier.java:92)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:206)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:201)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
	at org.eclipse.equinox.internal.event.EventAdminImpl.dispatchEvent(EventAdminImpl.java:131)
	at org.eclipse.equinox.internal.event.EventAdminImpl.sendEvent(EventAdminImpl.java:73)
	at org.eclipse.equinox.internal.event.EventComponent.sendEvent(EventComponent.java:44)
	at org.eclipse.e4.ui.services.internal.events.EventBroker.send(EventBroker.java:55)
	at org.eclipse.e4.ui.internal.workbench.UIEventPublisher.notifyChanged(UIEventPublisher.java:60)
	at org.eclipse.emf.common.notify.impl.BasicNotifierImpl.eNotify(BasicNotifierImpl.java:424)
	at org.eclipse.e4.ui.model.application.ui.impl.UIElementImpl.setToBeRendered(UIElementImpl.java:316)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.deleteModelElement(ModelServiceImpl.java:164)
	at org.eclipse.e4.ui.workbench.renderers.swt.StackRendererTest.cleanUp(StackRendererTest.java:99)
```

**Comment by vogella** [comment]
> Was it intended that this PR is merged despite test failures on MacOS? 

IIRC everything was green before the merge. Now it looks like this memory was is. We should obviously not merge PR with new test errors. I will revert

**Comment by vogella** [comment]
https://github.com/eclipse-platform/eclipse.platform.ui/pull/2348

**Comment by vogella** [comment]
Ah, these days Mac test results are out-of-sight. Sorry did not notice that.

image

**Comment by jukzi** [comment]
Depends on where you look. Somehow it's not consistent. Especially after rebuilding something i have seen test result was not updated appropriately sometimes.
image

**Pull request merged**
